crates/kubelet/src/handle.rs

```diff
@@ -33,27 +34,123 @@ pub trait Stop {
     async fn wait(&mut self) -> anyhow::Result<()>;
 }
 
+/// Trait to describe necessary behavior for creating multiple log readers.
+/// TODO: Both providers make a handle containing a tempfile. If this is a common pattern,
```

Comment:
Yep, I figured as much, do you think it is safe to say that all logging will be done to a file? The generic handle could just accept a path. Regardless, providers would still be free to implement the trait for whatever they are using.

Reply:
I'm not too sure on that assumption. One could write a provider that streams logs elsewhere. Having to avoid the middle-ground of writing to a file, then streaming that file somewhere else would be nice to avoid